rest-service/manager_rest/test/endpoints/test_events.py

```diff
@@ -121,89 +122,94 @@ def _populate_db(self):
                 workflow_id=fake.uuid4(),
-                _tenant_id=fake.uuid4(),
+                _tenant_id=deployment._tenant_id,
                 _creator_id=fake.uuid4(),
-                _deployment_fk=choice(deployments)._storage_id,
-            )
-            for _ in xrange(self.EXECUTION_COUNT)
-        ]
+                _deployment_fk=deployment._storage_id,
+            ))
         session.add_all(executions)
         session.commit()
 
-        nodes = [
-            Node(
+        nodes = []
+        for _ in xrange(self.NODE_COUNT):
+            deployment = choice(deployments)
+            nodes.append(Node(
                 id='node_{}'.format(fake.uuid4()),
                 deploy_number_of_instances=1,
                 max_number_of_instances=1,
                 min_number_of_instances=1,
                 number_of_instances=1,
                 planned_number_of_instances=1,
                 type='<type>',
-                _deployment_fk=choice(deployments)._storage_id,
-                _tenant_id=choice(deployments)._tenant_id,
-                _creator_id=choice(deployments)._creator_id,
-            )
-            for _ in xrange(self.NODE_COUNT)
-        ]
+                _deployment_fk=deployment._storage_id,
+                _tenant_id=deployment._tenant_id,
+                _creator_id=deployment._creator_id,
+            ))
         session.add_all(nodes)
         session.commit()
 
-        node_instances = [
-            NodeInstance(
+        node_instances = []
+        for _ in xrange(self.NODE_INSTANCE_COUNT):
+            node = choice(nodes)
+            node_instances.append(NodeInstance(
                 id='node_instance_{}'.format(fake.uuid4()),
                 state='<state>',
-                _node_fk=choice(nodes)._storage_id,
-                _tenant_id=choice(nodes)._tenant_id,
-                _creator_id=choice(nodes)._creator_id,
-            )
-            for _ in xrange(self.NODE_INSTANCE_COUNT)
-        ]
+                _node_fk=node._storage_id,
+                _tenant_id=node._tenant_id,
+                _creator_id=node._creator_id,
+            ))
         session.add_all(node_instances)
         session.commit()
 
         def create_event():
             """Create new event using the execution created above."""
+            execution = choice(executions)
             return Event(
                 id='event_{}'.format(fake.uuid4()),
                 timestamp=fake.date_time(),
                 reported_timestamp=fake.date_time(),
-                _execution_fk=choice(executions)._storage_id,
-                _tenant_id=choice(executions)._tenant_id,
-                _creator_id=choice(executions)._creator_id,
-                node_id=choice(node_instances).id,
+                _execution_fk=execution._storage_id,
+                _tenant_id=execution._tenant_id,
+                _creator_id=execution._creator_id,
+                node_id=choice(node_instances).id,  # TODO
```

Comment:
yep, turns out `.node_id` is actually used for node instances, see eg. https://github.com/cloudify-cosmo/cloudify-manager/blob/2a208f2cbb68b6c8f724a3321bfe55d751612fa5/rest-service/manager_rest/rest/resources_v1/events.py#L339